.dotnet/src/Custom/Files/FileClient.cs

```diff
@@ -178,12 +143,13 @@ public virtual ClientResult<BinaryData> DownloadFile(string fileId)
             throw new ClientResultException(message.Response);
         }
 
-        return ClientResult.FromValue(message.Response.Content, message.Response);
+        PipelineResponse response = message.ExtractResponse();
```

Comment:
When we do this, it becomes the end-user's responsibility to dispose the response and the content stream after use.  It would be factored a bit differently across convenience/protocol methods -- but this convenience method doesn't look like it calls through to a protocol method?  Is that intentional in the current implementation?

Reply:
I flagged this here -- let's loop back on it: https://github.com/joseharriaga/openai-in-typespec/pull/41/files#r1534791797

Reply:
Ahh, this is a very interesting problem -- it's exactly why we have `StreamingClientResult` currently implement `IDisposable` (so it can take full and unconditional ownership of the disposable unbuffered stream). One of the other options we initially considered was to have a result container object that provided "opt-in" access to the resource you need to manage yourself -- it wasn't fully clear how we'd make that play nicely without having the container *also* be IDisposable, but I'm sure there are some creative redirection approaches we could take.

Alternatively, we could generalize some concept of a "disposable client result" -- though that has its own baggage.

Reply:
Yeah, DisposableClientResult doesn't feel like a place we want to go .... I think maybe the best option is to "keep convenience methods convenient" -- @KrzysztofCwalina made a suggestion in the APIView that protocol methods can be the place end-users could get file outputs as Streams using the dual Content/ContentStream APIs on PipelineResponse.  There would be a need to implement such protocol/convenience methods differently so the ClientModel pipeline didn't buffer the response ContentStream in-memory, but the convenience method did ... 

I'll take this out of this PR since it was mostly here to see what it looked like, and so we can get the convenience methods for input merged, but I'll flag this for follow-up on the APIView comment.

Reply:
Resolving comment since the experimental "Stream output" approach has been removed from the PR.